module/zfs/zvol.c

```diff
@@ -1122,26 +1118,22 @@ zvol_create_minors_impl(const char *name)
 	list_create(&minors_list, sizeof (minors_job_t),
 	    offsetof(minors_job_t, link));
 
-	parent = kmem_alloc(MAXPATHLEN, KM_SLEEP);
-	(void) strlcpy(parent, name, MAXPATHLEN);
 
-	if ((atp = strrchr(parent, '@')) != NULL) {
+	if (strchr(name, '@') != NULL) {
 		uint64_t snapdev;
 
-		*atp = '\0';
-		error = dsl_prop_get_integer(parent, "snapdev",
+		int error = dsl_prop_get_integer(name, "snapdev",
 		    &snapdev, NULL);
 
 		if (error == 0 && snapdev == ZFS_SNAPDEV_VISIBLE)
 			error = ops->zv_create_minor(name);
```

Comment:
Do we need to do anything with these error values?

Reply:
Note that the one caller of zvol_create_minors_impl() discarded its return value.  I'm maintaining this semantic that minor node creation is "best effort", i.e. it fails silently (even though it should not fail as often now, and at least for the non-recursive cases it is synchronous and therefore we *could* report the error).

I'll change this to discard the return value here, rather than setting `error`, which is not used after this point.